python/shotgun_desktop/startup.py

```diff
@@ -775,6 +814,14 @@ def main(**kwargs):
             logger.info("Login canceled. Quitting.")
             return 0
 
+        # In the case where the site is using SSO, the user needs to renew
+        # its claims regularily. So we kick off a separate newewal thread.
+        if user._impl.get_saml_expiration() is not None:
```

Comment:
no - let's keep this clean pls - it should be refactored in some way or another so that we don't call the internals directly.